src/net.h

```diff
@@ -101,6 +101,17 @@ typedef int64_t NodeId;
 struct AddedNodeParams {
     std::string m_added_node;
     bool m_use_v2transport;
+    bool operator==(const AddedNodeParams& p) const
+    {
+        return m_added_node == p.m_added_node && m_use_v2transport == p.m_use_v2transport;
+    }
+};
+
+struct AddedNodeHasher {
+    size_t operator()(const AddedNodeParams& p) const noexcept
+    {
+        return std::hash<std::string>()(p.m_added_node) ^ (std::hash<bool>()(p.m_use_v2transport) << 1);
```

Comment:
I am not sure why they used `<< 1`. It seems to be useless and also it drops one bit from `h2`. Maybe to avoid returning `0` if first name and last name happen to be equal?

I think for the purposes of hashing `m_added_node_params` this suffices:
```cpp
    size_t operator()(const AddedNodeParams& p) const noexcept
    {
        return std::hash<std::string>()(p.m_added_node) ^ p.m_use_v2transport;
    }
```

However `std::hash` is not used anywhere in the code (modulo in some tests and benchmarks). `CServiceHash` uses `CSipHasher`, but there are security concerns with an outsider being able to fiddle with the hashing, maybe this is an overkill for `AddedNodeParams`. If that is to be used here, then it would look like:

```cpp
        CSipHasher hasher(m_salt_k0, m_salt_k1);
        hasher.Write(MakeUCharSpan(p.m_added_node));
        hasher.Write(p.m_use_v2transport);
        return static_cast<size_t>(hasher.Finalize());
```

Or use some of the stateless `Hash*` functions from `src/hash.h`.

I guess @sipa can give us an advice?